contrib/endpoints/src/api_manager/api_manager_impl.cc

```diff
@@ -75,11 +94,45 @@ utils::Status ApiManagerImpl::Init() {
     global_context_->cloud_trace_aggregator()->Init();
   }
 
-  for (auto it : service_context_map_) {
-    if (it.second->service_control()) {
-      it.second->service_control()->Init();
+  if (!service_context_map_.empty()) {
+    for (auto it : service_context_map_) {
+      if (it.second->service_control()) {
+        it.second->service_control()->Init();
+      }
+    }
+
+    if (global_context_->rollout_strategy() !=
+        std::string(kConfigRolloutStrategy)) {
+      return config_loading_status_;
     }
   }
+
+  config_manager_.reset(new ConfigManager(
+      global_context_,
+      [this](const utils::Status &status,
+             const std::vector<std::pair<std::string, int>> &configs) {
+        if (status.ok()) {
+          std::vector<std::pair<std::string, int>> rollouts;
+
+          for (auto item : configs) {
+            std::string config_id;
+            if (AddConfig(item.first, &config_id, true)) {
+              rollouts.push_back({config_id, item.second});
+            }
+          }
+
+          if (rollouts.size() == 0) {
+            config_loading_status_ =
+                utils::Status(Code::ABORTED, "Invalid service config");
+            return;
+          }
+
+          DeployConfigs(std::move(rollouts));
+        }
+        config_loading_status_ = status;
+      }));
+  config_manager_->Init();
```

Comment:
it is better to pass in callback function at Init() call.  Most likely,  onTimer() call needs different callback function.